C/plugins/storage/postgres/connection.cpp

```diff
@@ -3312,3 +3314,712 @@ const char *p;
+                               					                                 return false;
+                                                        				}
+                                                        				else
+                                                        				{
+												// keep the cols in indexes as a comma separated list of sorted columns
+				                                                                for (auto& i : v["index"].GetArray())
+                                                                				{
+                                                                        				indexVec.push_back(i.GetString());
+                                                                				}
+
+                                                                				std::sort(indexVec.begin(), indexVec.end());
+												for (auto i = 0; i < indexVec.size(); ++i)
+												{
+													s.append(indexVec[i]);
+													if (i < indexVec.size() -1){
+														s.append(",");
+													}
+												}
+                                                        				}
+                                                				}
+                                        				}
+									indexesMatrixFromReq.push_back(s);
+								}
+							}
+
+							// Traverse through the colums list found in DB for this table 
+							// and create/alter/delete the colums list
+							//
+
+							unordered_set<std::string> &dbCol = columnMapFromDB[name];
+							bool columnsToAlter = false;
+							for ( auto& v: colsPerTableInReq)
+							{
+								// table creation case
+								if (!alterTable)
+								{
+									sql += v.column + " " + v.type;
+									if (v.type == "varchar")
+									{
+										sql += "(" + std::to_string(v.sz) + ")";	
+									}
+
+									if (v.key == true)
+									{
+										sql += " primary key";
+									}
+									sql +=",";
+								}
+								else
+								{
+									// alter table case, table already exists
+									// check if column already exists in database
+									if (dbCol.find(v.column) == dbCol.end())
+									{
+										// if it is not a key then add the column else log error
+										if (!v.key)
+										{
+											sql += "add column ";
+											sql += v.column + " " + v.type;
+			                                                                if (v.type == "varchar")
+               				                                                {
+                        	                                                        	sql += "(" + std::to_string(v.sz) + ")";
+											}
+											sql +=",";
+											columnsToAlter = true;
+										}
+										else
+										{
+											// altering a key is not allowed
+											Logger::getLogger()->error("%s:%d altering key request(%s) is not allowed for an existing table", __FUNCTION__, __LINE__, v.column.c_str());
```

Comment:
Please give schema, service and table name

Reply:
Fixed